src/surge-fx/SurgeFXProcessor.cpp

```diff
@@ -411,6 +469,26 @@ void SurgefxAudioProcessor::processBlock(juce::AudioBuffer<float> &buffer,
             outR[i] = std::clamp(outR[i], -2.f, 2.f);
         }
     }
+
+    processBlockOSC();
+}
+
+// Pull incoming OSC events from ring buffer
+void SurgefxAudioProcessor::processBlockOSC()
+{
+    auto messages = oscRingBuf.popall();
```

Comment:
So `popall` is an allocating call. We should really do a while pop with a has value.

Also shouldn't this get short circuited with a

```
if (!oscReceiving)
   return;
```

If we add the second we can deal with the first after the fact but to do a vector creation on every block is probably not good! (And we should check if we do this without a short circuit in surge also).

Reply:
I used popall() in main Surge as well (it was suggested to me by Surgo, who wrote the RingBuffer). So if I change it here, I'll need to do so in SSP as well.

I believe that nothing will be pushed into the ring buffer is OSC is not receiving, so is that check necessary?

Reply:
I looked at what popall does and it will make a vector of size 0 in that case. If we know it will be empty not checking it would be smart. The popall vs repeated pop is less of an urgent issue but we could avoid a potentially unbounded alloc by doin git.

Reply:
OK, sounds like that should be changed both here and in SSP, then.

Reply:
If I commit changes to this PR, will it reintroduce the submodule issues? (thanks for cleaning that up!)